### PR TITLE
Avoid sending replies to topic exchange

### PR DESCRIPTION
**Pull request by indrekj**

### CODE REVIEW

File: lib/freddy.rb [hunk 1]
```diff
@@ -74,7 +74,7 @@ def respond_to(destination, &callback)
     @logger.info "Listening for requests on #{destination}"
 
     channel = @connection.create_channel
-    producer = Producers::SendAndForgetProducer.new(channel, @logger)
+    producer = Producers::ReplyProducer.new(channel, @logger)
```
Comment (deiwin) [inline lib/freddy.rb]: I'd maybe call the variable `reply_producer`, but as a name for this class it feels wrong.

Comment (deiwin) [inline lib/freddy.rb]: I take that back

**Comment by deiwin** [comment]
+1

**Comment by willmore** [comment]
No tests with a key change?

**Comment by indrekj** [comment]
@willmore Both solutions work correctly, the previous one just sent one message to a black hole. However you're right. Now if I think about it, this extra message logic could happen again when refactoring code. I'll add tests in the next PR.

**Comment by indrekj** [comment]
@willmore #35

**Pull request merged**